diff --git a/am335x-boneblack.dts.orig b/am335x-boneblack.dts
index cc702ed..3378a2a 100644
--- a/arch/arm/boot/dts/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/am335x-boneblack.dts
@@ -168,3 +168,28 @@
 		"NC",
 		"NC";
 };
+
+&am33xx_pinmux {
+    i2c1_pins: pinmux_i2c1_pins {
+        pinctrl-single,pins = <
+            AM33XX_PADCONF(AM335X_PIN_SPI0_D1, PIN_INPUT_PULLUP, MUX_MODE2)  /* spi0_d1.i2c1_sda, SLEWCTRL_SLOW | INPUT_PULLUP | MODE2 */
+            AM33XX_PADCONF(AM335X_PIN_SPI0_CS0, PIN_INPUT_PULLUP, MUX_MODE2) /* spi0_cs0.i2c1_scl, SLEWCTRL_SLOW | INPUT_PULLUP | MODE2 */
+        >;
+    };
+};
+
+&i2c1 {
+    status = "okay";
+    pinctrl-names = "default";
+    pinctrl-0 = <&i2c1_pins>;
+    
+    clock-frequency = <400000>;
+
+    #address-cells = <1>;
+    #size-cells = <0>;
+
+    ssd1306: oled@3c {
+        compatible = "hungpk,ssd1306-i2c";
+        reg = <0x3c>;
+    };
+};
